replication/row_event.go

```diff
@@ -342,18 +342,21 @@ func (e *RowsEvent) decodeValue(data []byte, tp byte, meta uint16) (v interface{
 		return nil, 0, nil
 	case MYSQL_TYPE_LONG:
 		n = 4
-		v = int64(binary.LittleEndian.Uint32(data))
+		v = int64(int32(binary.LittleEndian.Uint32(data)))
 	case MYSQL_TYPE_TINY:
 		n = 1
-		v = int64(data[0])
+		v = int64(int8(data[0]))
 	case MYSQL_TYPE_SHORT:
 		n = 2
-		v = int64(binary.LittleEndian.Uint16(data))
+		v = int64(int16(binary.LittleEndian.Uint16(data)))
 	case MYSQL_TYPE_INT24:
 		n = 3
-		v = int64(FixedLengthInt(data[0:3]))
+		u32 := uint32(FixedLengthInt(data[0:3]))
+		if u32 & 0x00800000 != 0 {
+			u32 |= 0xFF000000
+		}
+		v = int64(int32(u32))
 	case MYSQL_TYPE_LONGLONG:
-		//em, maybe overflow for int64......
 		n = 8
 		v = int64(binary.LittleEndian.Uint64(data))
```

Comment:
re-interpreting uint64 as int64 doesn't lose bits. later casting back to uint64 will give original unsigned numeric value, so comment safely removed.